action/envelope.go

```diff
@@ -28,18 +28,17 @@ type (
 	}
 
 	envelope struct {
-		version  uint32
-		chainID  uint32
-		nonce    uint64
-		gasLimit uint64
-		gasPrice *big.Int
-		payload  actionPayload
+		AbstractAction
```

Comment:
we should revert this change, which won't hurt this PR, and leave it to follow up PRs after discussion.